dhall/src/Dhall.hs

```diff
@@ -1659,6 +1690,9 @@ instance Inject a => Inject (Vector a) where
 instance Inject a => Inject (Data.Set.Set a) where
     injectWith = fmap (contramap Data.Set.toList) injectWith
 
+instance Inject a => Inject (Data.HashSet.HashSet a) where
```

Comment:
I wonder whether we should sort the elements… :thinking:

Reply:
In math, sets have no notion of ordering so I would tend towards not sorting it...

Reply:
I'm having second thoughts, maybe we should sort it... Extra opinion @Gabriel439?

Reply:
On second thought, this is something that we could change later on. For now, maybe just point out that there's no sorting, and add a doctest that demonstrates it.

Reply:
OK, I've done one for `HashSet` (not pushed yet), but what about `Set`? `Data.Set. toList ` and `toAscList` are both `O(n)`, I might as well use `toAscList` right?

Reply:
I think `Data.Set.toList` is defined via `toAscList`. `toAscList` is more explicit though. :+1:

Reply:
You are correct. Good.